src/commands/domain.rs

```diff
@@ -13,12 +14,33 @@ use crate::{
 
 use super::*;
 
-/// Generates a domain for a service if there is not a railway provided domain
-// Checks if the user is linked to a service, if not, it will generate a domain for the default service
+/// Add a custom domain or generate a railway provided domain for a service.
+///
+/// There is a maximum of 1 railway provided domain per service.
 #[derive(Parser)]
-pub struct Args {}
+pub struct Args {
+    /// The service to generate a domain for
+    #[clap(short, long)]
+    port: Option<u16>,
+
+    /// The name of the service to use
+    #[clap(short, long)]
+    service: Option<String>,
+
+    /// Optionally, specify a custom domain to use. If not specified, a domain will be generated.
+    ///
+    /// Specifying a custom domain will also return the required DNS records
+    /// to add to your DNS settings
+    domain: Option<String>,
+}
+
+pub async fn command(args: Args, json: bool) -> Result<()> {
+    if let Some(domain) = args.domain {
+        create_custom_domain(domain, args.port, args.service, json).await?;
```

Comment:
Can you also move the service domain creation to a `create_service_domain` function so it is clear what type of domain is being created.

Reply:
Ok